patcher/packages/mapPatcher/patcherExec.js

```diff
@@ -132,6 +132,106 @@ export function patcherExec(fileContents) {
         }
       }`;
     fileContents.GAMEMAIN = stringReplaceAt(fileContents.GAMEMAIN, startOfAirportsConfig, endOfAirportsConfig, newAirportsConfig);
+
+    
+    console.log("Modifying ocean depth layers");
+    const citiesWithOcean = config.places.filter(place => fs.existsSync(`${import.meta.dirname}/processed_data/${place.code}/ocean_depth_index.json`)).map(p => p.code);
+    const citiesWithOceanList = JSON.stringify(citiesWithOcean);
+    const vanillaCitiesList = "['ATL', 'AUS', 'BAL', 'BOS', 'CHI', 'CIN', 'CLE', 'CLT', 'DAL', 'DC', 'DEN', 'DET', 'HNL', 'HOU', 'IND', 'MIA', 'MSP', 'NYC', 'PDX', 'PHL', 'PIT', 'SAN', 'SEA', 'SF', 'SLC', 'STL']";
+
+    // Warn about cities without ocean data
+    config.places.forEach(place => {if (!citiesWithOcean.includes(place.code)) {console.log(`No ocean depth data found for ${place.name} (${place.code}). Ocean layers will be disabled.`);}});
+
+    fileContents.GAMEMAIN = fileContents.GAMEMAIN.replace(/const sources\s*=\s*\{/, `const sources = {...(${citiesWithOceanList}.indexOf(cityCode) > -1 ? { "ocean-depth-source": { type: "geojson", data: \`data/\${cityCode}_ocean.geojson\`, tolerance: 0 } } : {}),`);
+    fileContents.GAMEMAIN = fileContents.GAMEMAIN.replaceAll("showOceanFoundations: layersToShow.oceanFoundations", `showOceanFoundations: (${citiesWithOceanList}.indexOf(cityCode) > -1 || ${vanillaCitiesList}.indexOf(cityCode) > -1) ? layersToShow.oceanFoundations : !layersToShow.oceanFoundations`);
+
+    const oceanLayerBlock = `

```

Comment:
is the vanilla ocean layer not sufficient? just supply the depth index

Reply:
this may also end up overriding the vanilla one, just use the vanilla one and only supply ocean_depth_index.json